src/openapi/transport/auth.ts

```diff
@@ -0,0 +1,224 @@
+/**
+ * @module saxo/openapi/transport/auth
+ * @ignore
+ */
+
+// -- Local variables section --
+
+import log from '../../log';
+import type AuthProvider from '../authProvider';
+import TransportCore from './core';
+
+const LOG_AREA = 'TransportAuth';
+
+// The default period within which errors on different tokens
+// cause an endpoint auth errors to be ignored.
+const DEFAULT_AUTH_ERRORS_DEBOUNCE_PERIOD = 30000; // ms
+
+// -- Local methods section --
+
+type Methods = 'get' | 'put' | 'post' | 'delete' | 'patch' | 'options' | 'head';
+
+type Options = {
+    authErrorsDebouncePeriod?: number;
+};
+
+function makeTransportMethod(this: TransportAuth, method: Methods) {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const that = this;

```

Comment:
I think we don't need to make such tricks, what about typing `this` just in the returned function and use it normally?